server/src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -698,9 +699,41 @@ public String toString() {
         }
     }
 
-    public static class Index extends Timer implements ToXContentFragment, Writeable {
-
+    public static class RecoveryFilesDetails {
         private final Map<String, File> fileDetails = new HashMap<>();
+
+        public File addFileDetails(String name, File file) {
```

Comment:
Can we instantiate the File - and check that it does not previously existed - directly within this method?